server/src/main/java/org/elasticsearch/index/mapper/vectors/VectorDVLeafFieldData.java

```diff
@@ -76,4 +86,94 @@ public DocValuesScriptFieldFactory getScriptFieldFactory(String name) {
+                            return false;
+                        }
+                        BytesRef ref = binary.binaryValue();
+                        ByteBuffer byteBuffer = ByteBuffer.wrap(ref.bytes, ref.offset, ref.length);
+                        if (indexVersion.onOrAfter(DenseVectorFieldMapper.LITTLE_ENDIAN_FLOAT_STORED_INDEX_VERSION)) {
+                            byteBuffer.order(ByteOrder.LITTLE_ENDIAN);
+                        }
+                        values = new byte[dims];
+                        for (int dim = 0; dim < dims; dim++) {
+                            values[dim] = byteBuffer.get();
+                        }
+                    }
+                    valuesCursor = 0;
+                    return true;
+                }
+
+                public Object nextValue() {
+                    return values[valuesCursor++];
+                }
+            };
+            case FLOAT -> new DenseVectorNumericValues(dims) {
+                private float[] values;
+                private FloatVectorValues floatVectorValues; // use when indexed
+
+                @Override
+                public boolean advanceExact(int docId) throws IOException {
+                    if (indexed) {
+                        if (floatVectorValues == null) {
+                            this.floatVectorValues = reader.getFloatVectorValues(field);
+                        }
+                        if (iterator == null) {
+                            this.iterator = floatVectorValues.iterator();
+                        }
+                        if (iterator.advance(docId) == NO_MORE_DOCS) {
+                            return false;
+                        }
+                        values = floatVectorValues.vectorValue(iterator.index());
+                    } else {
+                        if (binary == null) {
+                            binary = DocValues.getBinary(reader, field);
+                        }
+                        if (binary.advance(docId) == NO_MORE_DOCS) {
+                            return false;
+                        }
+                        BytesRef ref = binary.binaryValue();
+                        ByteBuffer byteBuffer = ByteBuffer.wrap(ref.bytes, ref.offset, ref.length);
+                        if (indexVersion.onOrAfter(DenseVectorFieldMapper.LITTLE_ENDIAN_FLOAT_STORED_INDEX_VERSION)) {
+                            byteBuffer.order(ByteOrder.LITTLE_ENDIAN);
+                        }
+                        values = new float[dims];
+                        for (int dim = 0; dim < dims; dim++) {
+                            values[dim] = byteBuffer.getFloat();
+                        }
+                    }
+                    valuesCursor = 0;
+                    return true;
+                }
+
+                @Override
+                public Object nextValue() {
+                    return values[valuesCursor++];
```

Comment:
We don't need `valueCursor`. I **think** `nextValue` can always return the same value

```java
public Object nextValue() throws IOException {
      return values;
}
```

Reply:
Same issue as `docValueCount` returning 1.